channel.go

```diff
@@ -168,6 +168,9 @@ func (ch *Channel) open() error {
 // Performs a request/response call for when the message is not NoWait and is
 // specified as Synchronous.
 func (ch *Channel) call(req message, res ...message) error {
+	ch.m.Lock()
```

Comment:
Isn't there a lock implemented on the connection object `c.sendM` before writing to the frame. That should be doing the same thing, right?

Reply:
This is indeed typically done at connection level by various clients. We have to be really careful here as throughput effects of locking on the hot path can be very serious or even catastrophic for some systems.